profilingdemo/source/java/src/org/kaaproject/kaa/demo/profiling/ProfilingDemo.java

```diff
@@ -0,0 +1,198 @@
+
+        // Create the Kaa desktop context for the application.
+        DesktopKaaPlatformContext desktopKaaPlatformContext = new DesktopKaaPlatformContext(kaaClientProperties);
+
+        // Create a Kaa client and add a listener which displays the Kaa client profiling
+        // as soon as the Kaa client is started.
+        final KaaClient kaaClient = Kaa.newClient(desktopKaaPlatformContext, new SimpleKaaClientStateListener() {
+            @Override
+            public void onStarted() {
+                super.onStarted();
+
+                displayConfiguration(index);
+            }
+        }, true);
+        kaaClients.add(kaaClient);
+
+        kaaClient.setProfileContainer(new ProfileContainer() {
+            @Override
+            public PagerClientProfile getProfile() {
+                InnerData data = init().get(index);
+
+                PagerClientProfile pagerClientProfile = new PagerClientProfile();
+                pagerClientProfile.setAudioSupport(data.isAudio());
+                pagerClientProfile.setVibroSupport(data.isVibro());
+                pagerClientProfile.setVideoSupport(data.isVideo());
+
+                return pagerClientProfile;
+            }
+        });
+
+        // Persist profiling in a local storage to avoid downloading it each time the Kaa client is started.
+        kaaClient.setConfigurationStorage(new SimpleConfigurationStorage(desktopKaaPlatformContext, "saved_config.cfg"));
+
+        // Add a listener which displays the Kaa client profiling each time it is updated.
+        kaaClient.addConfigurationListener(new ConfigurationListener() {
+            @Override
+            public void onConfigurationUpdate(PagerConfiguration configuration) {
+                displayConfiguration(index);
+            }
+        });
+
+        // Start the Kaa client and connect it to the Kaa server.
+        kaaClient.start();
+    }
+
+    public static void stopKaaClients() throws IOException {
+        LOG.info("Stopping clients...");
+        for (KaaClient k : kaaClients) {
+            if (k != null)
+                k.stop();
+        }
+    }
+
+    public static void displayConfiguration(int index) {
+        if (index >= kaaClients.size()) {
+            LOG.error("Index more than KaaClient's list size!");
+            return;
+        }
+
+        KaaClient kaaClient = kaaClients.get(index);
+        if (kaaClient != null) {
+            PagerConfiguration configuration = kaaClient.getConfiguration();
+
+            boolean isAudioSupport = configuration.getAudioSubscriptionActive();
+            boolean isVideoSupport = configuration.getVideoSubscriptionActive();
+            boolean isVibroSupport = configuration.getVibroSubscriptionActive();
+
+            LOG.info("KeyHash - " + kaaClient.getEndpointKeyHash());
+
+            LOG.info(index + "-th profiling body (have vibro-, audio-, video-support):");
+            LOG.info("{} - {} - {}", isVibroSupport, isAudioSupport, isVideoSupport);
```

Comment:
I propose to add client profile into clients output before configuration, it will help the user to "link" which profile particular configuration correspond. And format message that it would more comfort to read. And use next property order: audio-, video-, vibration- as in a profile schema.